src/Microsoft.PowerShell.Commands.Utility/commands/utility/WaitEventCommand.cs

```diff
@@ -47,16 +47,34 @@ public int Timeout
         {
             get
             {
-                return _timeoutInSeconds;
+                return (int)_timeoutTimespan.TotalSeconds;
             }
 
             set
             {
-                _timeoutInSeconds = value;
+                _timeoutTimespan = TimeSpan.FromSeconds(value);
             }
         }
 
-        private int _timeoutInSeconds = -1; // -1: infinite, this default is to wait for as long as it takes.
+        /// <summary>
+        /// If timespan is specified, the cmdlet will only wait for this timespan.
+        /// Negative values mean never timeout.
+        /// </summary>
+        [Parameter]        
+        public TimeSpan TimeSpan
```

Comment:
I guess we could change type of Timeout parameter to TimeSpan and add type converter (seconds to TimeSpan) to mitigate the breaking change

Reply:
It'd be nice if we could, and maybe I'm missing some type converter magic we could use.

When casting [Timespan] from an [int], it casts to ticks, not seconds.

Under the covers I'm turning -TimeOut into -Timespan.

One other thought to avoid having a new parameter would be having -TimeOutSec become a [double], which would allow us to pass, say 0.5.

Unfortunately, I believe this is likely to create some accidental back-compat errors, as scripts with a timeout of 0.25 would run fine on later versions and not wait at all on earlier ones (because of rounding).

IMHO, having a new parameter is a much better hint that it's a newer version.

Reply:
> IMHO, having a new parameter is a much better hint that it's a newer version.

If we can use an existing parameter, we should do so. Otherwise it will mislead users, it's bad UX.

PowerShell can do clever things with parameters. You can find examples by searching for ArgumentTransformationAttribute.

Reply:
@SteveL-MSFT what's your opinion:  would you rather a new parameter or transform the old one?

Reply:
Let me bring this up to the Cmdlets WG

Reply:
@SteveL-MSFT / the working group:

I'd like to see this fix go in, as it would fairly immediately improve the performance of working modules.

I'm fairly confident that having two (or more) parameters would be the right way to go here.  To restate the reasons:

* Making -TimeoutSec a `[double]` would break back-compat in a weird way, due to rounding.
* Making -TimeOutSet a `[TimeSpan]` would create weird casting errors, as ints become 'ticks' in [TimeSpan]s
* Making something more complicated to make [TimeSpan] magically castable from [int] might have been nice if it was PowerShell 1, but, as it stands, adding something like a type converter here would potentially break a lot of people's code (and complicate this pull request).
* Adding a -CancellationToken would be desirable, and I believe it should be another PR.

I am trying to keep this PR concise, as I expect to do future work in this space, and I am trying to do one PR per issue.

If there is a solid blocking reason to pend on this, please let me know what it might be, and I'll be happy to make any modifications you need.


To restate the rationale, in the hopes of a speedy resolution:

_No one should have to wait a second to find out things didn't work._ The faster one can time out a wait for an event that should be effectively immediate, the faster the rest of the code will work.

Please let me know if any adjustments are needed.

Hope this helps,

James